app/src/main/java/org/astraea/partitioner/smoothPartitioner/SmoothWeightPartitioner.java

```diff
@@ -216,12 +225,16 @@ private boolean isTransitionValid(
     }
 
     private void transitionTo(State target) {
-      if (!currentState.isTransitionValid(currentState, target)) {
-        throw new KafkaException(
-            "Invalid transition attempted from state "
-                + currentState.name()
-                + " to state "
-                + target.name());
+      try {
+        if (!currentState.isTransitionValid(currentState, target)) {
+          throw new KafkaException(
+              "Invalid transition attempted from state "
+                  + currentState.name()
+                  + " to state "
+                  + target.name());
+        }
+      } catch (KafkaException e) {
+        System.out.println(e);
```

Comment:
just swallow exception?

Reply:
I will open a special PR to deal with the issue of dependency, here I will change it to TODO first.

Reply:
Please file issue or attach link here

Reply:
OK in #190